File: interview/interview/custom/python/event.py
import frappe
@frappe.whitelist()
def update_ji(subject,event,date,job,round):
    data = frappe.get_doc("Job Interview",job)
    for row in data.rounds:
        if row.rounds == round:
            row.subject = subject
            row.scheduled_date = date
            row.event = event
        data.save()
        
def validate(self,action):
    if not self.event_participants and self.job_interview_name:
        frappe.throw("Event Participants should not be empty")

File: interview/interview/custom/js/event.js
frappe.ui.form.on("Event",{
    after_save: function(frm){
        frappe.call({
            method: 'interview.interview.custom.python.event.update_ji',
            args:{
                subject : frm.doc.subject,
                event: frm.doc.name,
                date: frm.doc.starts_on,
                job:frm.doc.job_interview_name,
                round: frm.doc.round
            }
        })
    },
    refresh: function(frm){
        if(! frm.doc.__islocal && frm.doc.job_interview_name){
            frm.add_custom_button(__("Go to Job Interview"),() => {
                frappe.set_route("Form",'Job Interview',frm.doc.job_interview_name)
            })
        }
    },
})